### PR TITLE
Feature/loss function

### PR DESCRIPTION
**Pull request by atsushi-green**

### CODE REVIEW

File: scripts/StationData.py [hunk 1]
```diff
@@ -81,25 +82,26 @@ def calc_graph_distance(self, station_df: pd.DataFrame, edge_df: pd.DataFrame, s
...
         edge_attr = []
         edge_list = []
-
         for i in range(len(station2id)):
             for j in range(len(station2id)):
                 try:
                     distance = dijkstra_dict[i][0][j]  # iを出発点としたときのjまでの距離(0番目の要素)
                     hop = len(dijkstra_dict[i][1][j]) - 1  # iを出発点としたときのjまでのホップ数(1番目の要素のリスト長-1)
-                    if hop > 2:
+                    if hop > 5:
```
Comment (atsushi-green) [inline scripts/StationData.py:104]: マジックナンバー

File: scripts/VariationalGraohAutoEncoder.py [hunk 2]
```diff
@@ -1,19 +1,28 @@
...
-        self.conv1 = GCNConv(in_channels, hidden_channels_list[0])
+        # self.conv1 = GCNConv(in_channels, hidden_channels_list[0])
+        self.conv1 = GATConv(in_channels, hidden_channels_list[0])
+
+        # self.conv_list = [
+        #     GCNConv(hidden_channels_list[i], hidden_channels_list[i + 1]) for i in range(len(hidden_channels_list) - 1)
+        # ]
         self.conv_list = [
-            GCNConv(hidden_channels_list[i], hidden_channels_list[i + 1]) for i in range(len(hidden_channels_list) - 1)
+            GATConv(hidden_channels_list[i], hidden_channels_list[i + 1]) for i in range(len(hidden_channels_list) - 1)
```
Comment (atsushi-green) [inline scripts/VariationalGraohAutoEncoder.py]: SAGEConvが良さそう

File: scripts/main_vgae.py [hunk 3]
```diff
@@ -7,15 +7,38 @@
 from torch import nn
 from torch_geometric.nn import VGAE
 from tqdm import tqdm
-from VariationalGraohAutoEncoder import VariationalGraohAutoEncoder
+from VariationalGraohAutoEncoder import VariationalGraohAutoDecoder, VariationalGraohAutoEncoder
 
-HIDDIN_DIM_LIST: final = [20, 15, 10]
+EMBEDDING_DIM = 5
+HIDDIN_DIM_LIST: final = [10, 10, 10, 10, 10, 10, 10, 10]
```
Comment (atsushi-green) [inline scripts/main_vgae.py:13]: SAGEConvなら多層化しても良さそう

File: scripts/main_vgae.py [hunk 4]
```diff
@@ -26,39 +49,36 @@ def main():
...
     optimizer = torch.optim.Adam(model.parameters(), lr=0.01)
 
     # 学習
-    for _ in tqdm(range(1, 1001)):
-        train(model, optimizer, data)
-    model.eval()
+    for epoch in tqdm(range(1, 100001)):
+        loss = train(model, optimizer, data)
+        if epoch % 100 == 0:
+            # train_loss, val_loss, test_loss = test(model, data, loss_function)
```
Comment (atsushi-green) [inline scripts/main_vgae.py:62]: 後でval, testの損失も計算させる
